src/test/bridges/angle/Angle.t.sol

```diff
@@ -0,0 +1,275 @@
+            sanDaiAsset,
+            emptyAsset,
+            daiAsset,
+            emptyAsset,
+            uint256(_amount),
+            0,
+            1,
+            address(0)
+        );
+        assertEq(IERC20(sanDaiAsset.erc20Address).balanceOf(address(bridge)), DUST);
+        assertEq(IERC20(daiAsset.erc20Address).balanceOf(address(bridge)), outputValueA + DUST);
+        (, , , , , uint256 sanRate, , , ) = bridge.STABLE_MASTER().collateralMap(
+            0xc9daabC677F3d1301006e723bD21C60be57a5915
+        );
+        assertEq(outputValueA, ((uint256(_amount) - DUST) * sanRate) / 1e18 - DUST);
+
+        uint256 input2 = 5 * uint256(_amount);
+        deal(sanDaiAsset.erc20Address, address(bridge), input2);
+        deal(daiAsset.erc20Address, address(bridge), 0); // reset DAI balance
+        (outputValueA, , ) = bridge.convert(sanDaiAsset, emptyAsset, daiAsset, emptyAsset, input2, 0, 1, address(0));
+        assertEq(IERC20(sanDaiAsset.erc20Address).balanceOf(address(bridge)), DUST);
+        assertEq(IERC20(daiAsset.erc20Address).balanceOf(address(bridge)), outputValueA + DUST);
+        (, , , , , sanRate, , , ) = bridge.STABLE_MASTER().collateralMap(0xc9daabC677F3d1301006e723bD21C60be57a5915);
+        assertEq(outputValueA, ((input2 - DUST) * sanRate) / 1e18 - DUST);
+
+        vm.stopPrank();
+    }
+
+    function testValidDepositE2E(uint96 _amount) public {
+        vm.assume(_amount >= 10);
+        uint256 balanceRollupBeforeDAI = IERC20(daiAsset.erc20Address).balanceOf(address(ROLLUP_PROCESSOR));
+        uint256 amount = uint256(_amount);
+        vm.assume(amount < balanceRollupBeforeDAI);
+
+        uint256 balanceRollupBeforeSanDAI = IERC20(sanDaiAsset.erc20Address).balanceOf(address(ROLLUP_PROCESSOR));
+
+        uint256 bridgeCallData = encodeBridgeCallData(id, daiAsset, emptyAsset, sanDaiAsset, emptyAsset, 0);
+        (uint256 outputValueA, , ) = sendDefiRollup(bridgeCallData, amount);
+
+        (, , , , , uint256 sanRate, , , ) = bridge.STABLE_MASTER().collateralMap(
+            0xc9daabC677F3d1301006e723bD21C60be57a5915
+        );
+
+        assertEq(outputValueA, ((amount - DUST) * 1e18) / sanRate - DUST);
+        assertEq(IERC20(daiAsset.erc20Address).balanceOf(address(bridge)), DUST);
+        assertEq(IERC20(sanDaiAsset.erc20Address).balanceOf(address(bridge)), DUST);
+
+        uint256 balanceRollupAfterDAI = IERC20(daiAsset.erc20Address).balanceOf(address(ROLLUP_PROCESSOR));
+        uint256 balanceRollupAfterSanDAI = IERC20(sanDaiAsset.erc20Address).balanceOf(address(ROLLUP_PROCESSOR));
+
+        assertEq(balanceRollupAfterDAI, balanceRollupBeforeDAI - amount);
+        assertEq(balanceRollupAfterSanDAI, balanceRollupBeforeSanDAI + outputValueA);
+    }
+
+    function testValidWithdrawE2E(uint96 _amount) public {
+        // the protocol might not have so many sanDAI available, so we limit what can be withdrawn
+        vm.assume(_amount >= 10 && _amount < 100000 ether);
```

Comment:
Here would be a good idea to use `bound(...)` method from _forge-std_ instead of `vm.assume(...)` because `vm.assume(...)` makes too many of the fuzz runs skip the test here (range is limited too much). We use bound a lot in other tests so feel free to check it out.